numpy/core/arrayprint.py

```diff
@@ -743,20 +755,22 @@ def __init__(self, precision, sign=False):
 
     def __call__(self, x):
         if isnan(x):
-            if self.sign:
-                return '+' + _nan_str
+            nan_str = _format_options['nanstr']
+            if self.sign == '+':
+                return '+' + nan_str
             else:
-                return ' ' + _nan_str
+                return ' ' + nan_str
         elif isinf(x):
+            inf_str = _format_options['infstr']
             if x > 0:
-                if self.sign:
-                    return '+' + _inf_str
+                if self.sign == '+':
+                    return '+' + inf_str
                 else:
-                    return ' ' + _inf_str
+                    return ' ' + inf_str
```

Comment:
Should we not omit the space for `sign=='-'`? Solution might be the same as the one suggested for float above. Or was this part of the "let's not touch long yet..."?